physics/apsides_body.hpp

```diff
@@ -135,6 +137,115 @@ void ComputeApsides(Trajectory<Frame> const& reference,
   }
 }
 
+template<typename Frame>
+std::vector<Interval<Instant>> ComputeCollisionIntervals(
+    RotatingBody<Frame> const& reference_body,
+    Trajectory<Frame> const& reference,
+    Trajectory<Frame> const& trajectory,
+    DiscreteTrajectory<Frame> const& apoapsides,
+    DiscreteTrajectory<Frame> const& periapsides) {
+  auto squared_distance_from_centre = [&reference,
+                                       &trajectory](Instant const& time) {
+    Position<Frame> const body_position =
+        reference.EvaluatePosition(time);
+    Position<Frame> const trajectory_position =
+        trajectory.EvaluatePosition(time);
+    Displacement<Frame> const displacement =
+        trajectory_position - body_position;
+    return displacement.Norm²();
+  };
+
+  auto const max_radius² = Pow<2>(reference_body.max_radius());
+
+  // NOTE(phl): This function doesn't consider |t_min| and |t_max| as potential
+  // periapsis, only as potential apoapsis.  Unsure if that's right.
```

Comment:
That cannot be right.
In particular a trajectory could have neither apoapsis nor periapsis (it could be monotone in altitude from t_min to t_max), and still run into the side of a mountain.

For the purposes of this function, t_min and t_max should always be treated as apsides of the kind opposite the first actual apsis and last actual apsis respectively.

Reply:
Done, PTAL.